app/src/main/java/de/westnordost/streetcomplete/data/osm/edits/update_tags/StringMapChangesBuilder.kt

```diff
@@ -1,68 +1,81 @@
 package de.westnordost.streetcomplete.data.osm.edits.update_tags
 
-class StringMapChangesBuilder(private val source: Map<String, String>) {
+class StringMapChangesBuilder(private val source: Map<String, String>): Map<String, String> {
     private val changes: MutableMap<String, StringMapEntryChange> = mutableMapOf()
 
-    fun delete(key: String) {
-        val valueBefore = requireNotNull(source[key]) { "The key '$key' does not exist in the map '$source'." }
-        val change = StringMapEntryDelete(key, valueBefore)
-        if (changes[key] == change) return
-        checkDuplicate(key)
-        changes[key] = change
-    }
-
-    fun deleteIfExists(key: String) {
-        if (source[key] != null) {
-            delete(key)
+    /** Remove the given key from the map */
```

Comment:
```suggestion
    /** Remove the given [key] from the map */
```

Reply:
Is this javadoc syntax? This is not used (yet) in the codebase for documentation comments. But if it works like that, it may be worth adding it everywhere for better code navigation and highlighting. But then it should be a separate PR so that it is done consistently

Reply:
No, it's Kotlin's equivalent, KDoc:

> The language used to document Kotlin code (the equivalent of Java's JavaDoc) is called KDoc. In its essence, KDoc combines JavaDoc's syntax for block tags (extended to support Kotlin's specific constructs) and Markdown for inline markup.

The syntax being used here is documented here: https://kotlinlang.org/docs/kotlin-doc.html#inline-markup
The style guide also mentions it: https://kotlinlang.org/docs/coding-conventions.html#documentation-comments

I thought I had seen it in the repo recently, but maybe I'm wrong. Postponing this for another PR is a good option then.